docker-app/worker_wrapper/wrapper.py

```diff
@@ -266,6 +262,10 @@ def _run_docker(
             cpu_shares=config.WORKER_QGIS_CPU_SHARES,
         )
 
+        # `docker_started_at`/`docker_finished_at` tracks the time spent on docker only
+        self.job.docker_started_at = timezone.now()
+        self.job.container_id = container.id
+        self.job.save()
```

Comment:
Hm, are you sure? In my understanding before it worked like:

1) `docker_started_at = now()`
2) initiate starting the container in detached mode. The container must have started when the line is finished.
3) block the python code until it finished via `container.wait()`
4) `docker_finished_at = now()`

While in your new case you swapped 1) and 2).

Reply:
I thought as the container.run command returns immediately it would not matter so much and just wanted to group the job.save() (could use also `update_fields`). But can revert this.

Reply:
It is not precisely immediately. It still call the docker daemon and might fail or not.

Reply:
Yep thought if it fails to start/run it is not wrong have NULL there?

Reply:
Absolutely.

Reply:
Probably I got the wrong impression, but I thought you were about to move `job.docker_started_at` back before calling `client.containers.run()`? I guess `job.container_id` should stay where it is.

Reply:
```suggestion
        # `docker_started_at`/`docker_finished_at` tracks the time spent on docker only
        self.job.docker_started_at = timezone.now()
        self.job.container_id = container.id
        self.job.save(update_fields=["docker_started_at", "container_id"])
```

Reply:
ok, sure